### PR TITLE
DOC: clarify that set_coordinates modifies array of geometries in place

### LINKED ISSUE #94: BUG: get/set_coordinates modifies the original geometries
The `set_coordinates` docstring indicates that it "Returns a copy of a geometry array with different coordinates."

However, this doesn't seem to be working correctly:

```In [31]: arr = pygeos.points([1, 2, 3], [1, 2, 3])

In [32]: arr
Out[32]: 
array([<pygeos.Geometry POINT (1 1)>, <pygeos.Geometry POINT (2 2)>,
       <pygeos.Geometry POINT (3 3)>], dtype=object)

In [33]: coords = pygeos.get_coordinates(arr) 

In [35]: new_coords = coords * 2 

In [36]: arr2 = pygeos.set_coordinates(arr, new_coords) 

In [37]: arr2  
Out[37]: 
array([<pygeos.Geometry POINT (2 2)>, <pygeos.Geometry POINT (4 4)>,
       <pygeos.Geometry POINT (6 6)>], dtype=object)

In [39]: arr  
Out[39]: 
array([<pygeos.Geometry POINT (2 2)>, <pygeos.Geometry POINT (4 4)>,
       <pygeos.Geometry POINT (6 6)>], dtype=object)
```

where the original array was modified in place.

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #94

### CODE REVIEW

File: pygeos/coordinates.py [hunk 1]
```diff
@@ -136,18 +136,29 @@ def get_coordinates(geometry, include_z=False, return_index=False):
...
+    """Sets the coordinates of a geometry array with different coordinates.
 
     If the coordinates array has shape (N, 2), all returned geometries
     will be two-dimensional, and the third dimension will be discarded,
     if present. If the coordinates array has shape (N, 3), the returned
     geometries preserve the dimensionality of the input geometries.
 
+    .. warning::
+
+        The geometries are modified in place! If you don't want to
```
Comment (caspervdw) [inline pygeos/coordinates.py:148]: Not exactly true, the geometries are not modified inplace, but the *array* is. The geometries in the array are replaced with new ones. So the geometry itself is immutable.

```suggestion
        The geometry array is modified in place! If you don't want to
```

File: pygeos/coordinates.py [hunk 2]
```diff
@@ -136,18 +136,29 @@ def get_coordinates(geometry, include_z=False, return_index=False):
 
 
 def set_coordinates(geometry, coordinates):
-    """Returns a copy of a geometry array with different coordinates.
+    """Sets the coordinates of a geometry array with different coordinates.
```
Comment (caspervdw) [inline pygeos/coordinates.py:139]:
```suggestion
    """Adapts the coordinates of a geometry array in-place.
```

**Review by caspervdw — CHANGES_REQUESTED** [review]
Thanks for this fix, I added some suggestions.

File: pygeos/coordinates.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pygeos/coordinates.py]: Ah, yes, that was indeed incorrectly worded. Thanks for the edits!

**Review by caspervdw — APPROVED** [review]

File: pygeos/coordinates.py [hunk 3]
```diff
@@ -136,18 +136,29 @@ def get_coordinates(geometry, include_z=False, return_index=False):
...
+    """Adapts the coordinates of a geometry array in-place.
 
     If the coordinates array has shape (N, 2), all returned geometries
     will be two-dimensional, and the third dimension will be discarded,
     if present. If the coordinates array has shape (N, 3), the returned
     geometries preserve the dimensionality of the input geometries.
 
+    .. warning::
+
+        The geometry array is modified in place! If you don't want to
```
Comment (caspervdw) [inline pygeos/coordinates.py:148]:
```suggestion
        The geometry array is modified in-place! If you do not want to
```

**Pull request merged**